src/v/cluster/partition_manager.h

```diff
@@ -33,7 +34,7 @@ class partition_manager
   : public ss::peering_sharded_service<partition_manager> {
 public:
     using ntp_table_container
-      = absl::flat_hash_map<model::ntp, ss::lw_shared_ptr<partition>>;
+      = model::ntp_flat_map_type<ss::lw_shared_ptr<partition>>;
```

Comment:
This map typedef just gives you the "best" flat map type for ntp, with heterogeneous lookup for ktp without needing to mess with comparators yourself.